EventBrite/Drivers/EventBriteFieldDisplayDriver.cs

```diff
@@ -0,0 +1,188 @@
+
+                model.Data = JsonConvert.SerializeObject(field.EventBriteData == null ? new EventBriteItem() : field.EventBriteData);
+            });
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(EventBriteField field, IUpdateModel updater, UpdateFieldEditorContext context)
+        {
+            var model = new EditEventBriteFieldViewModel();
+
+            await updater.TryUpdateModelAsync(model, Prefix, m => m.Data);
+
+            try
+            {
+                var settings = await _eventBriteSettingsService.GetSettingsAsync();
+
+                if (string.IsNullOrWhiteSpace(settings.PrivateToken))
+                {
+                    updater.ModelState.AddModelError("Value", "Eventbrite API settings are not set correctly, please check these and try again.");
+                }
+
+                var url = GetUrl(updater, model);
+
+                var eventBriteEventDto = await GetEventBriteEventDtoAsync(updater, settings, url);
+
+                EventBriteVenueDto eventBriteVenueDto = await GetEventBriteVenueDtoAsync(updater, settings, eventBriteEventDto);
+
+                if (updater.ModelState.ErrorCount > 0)
+                {
+                    return Edit(field, context);
+                }
+
+                if (eventBriteVenueDto != null)
+                {
+                    field.EventBriteData = new EventBriteItem(eventBriteEventDto, eventBriteVenueDto);
+                }
+                else
+                {
+                    field.EventBriteData = new EventBriteItem(eventBriteEventDto);
+                }
+            }
+            catch (Exception e)
+            {
+                updater.ModelState.AddModelError("Value", "Something went wrong saving the event");
+            }
+
+            return Edit(field, context);
+        }
+
+        private async Task<EventBriteEventDto> GetEventBriteEventDtoAsync(IUpdateModel updater, EventBriteSettings settings, string url)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Add("Authorization", string.Format("Bearer {0}", settings.PrivateToken));
+
+            var client = _clientFactory.CreateClient();
+
+            var response = await client.SendAsync(request);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                updater.ModelState.AddModelError("Value", "Error fetching event from EventBrite API, please check the Event ID and API Credentials");
+                return null;
+            }
+            return JsonConvert.DeserializeObject<EventBriteEventDto>(await response.Content.ReadAsStringAsync());
+        }
+
+        private async Task<EventBriteVenueDto> GetEventBriteVenueDtoAsync(IUpdateModel updater, EventBriteSettings settings, EventBriteEventDto eventBriteEventDto)
+        {
+            if (!string.IsNullOrWhiteSpace(eventBriteEventDto.venue_id))
```

Comment:
Avoid long `if` blocks. Why not switch this to be the other way round so the method starts.

```
if (string.IsNullOrWhiteSpace(eventBriteEventDto.venue_id)) {
    return null;
}
```

That way you avoid large logical blocks.